src/chainparams.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2020 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#pragma once
+
+#include "chainparamsbase.h"
+#include "consensus/params.h"
+#include "protocol.h"
+
+#include <memory>
+#include <vector>
+
+
+typedef std::map<int, uint256> MapCheckpoints;
+
+struct CCheckpointData {
+    MapCheckpoints mapCheckpoints;
+};
+
+/**
+ * CChainParams defines various tweakable parameters of a given instance of the
+ * Gridcoin system. There are two: the main network on which people trade goods
+ * and services and the public test network
+ */
+class CChainParams
+{
+public:
+    enum Base58Type {
+        PUBKEY_ADDRESS,
+        SCRIPT_ADDRESS,
+
+        MAX_BASE58_TYPES
+    };
+
+    const Consensus::Params& GetConsensus() const { return consensus; }
+    const CMessageHeader::MessageStartChars& MessageStart() const { return pchMessageStart; }
+    const std::vector<unsigned char>& AlertKey() const { return vAlertPubKey; }
+    int GetDefaultPort() const { return nDefaultPort; }
+
+    // const CBlock& GenesisBlock() const { return genesis; }
+    /** If this chain is exclusively used for testing */
+    bool IsTestChain() const { return m_is_test_chain; }
+    /** If this chain allows time to be mocked */
+    bool IsMockableChain() const { return m_is_mockable_chain; }
+    /** Minimum free space (in GB) needed for data directory */
+    uint64_t AssumedBlockchainSize() const { return m_assumed_blockchain_size; }
+    /** Return the network string */
+    std::string NetworkIDString() const { return strNetworkID; }
+    const unsigned char& Base58Prefix(Base58Type type) const { return base58Prefix[type]; }
+    const CCheckpointData& Checkpoints() const { return checkpointData; }
```

Comment:
I think that we may want to hook this up in checkpoints.cpp and remove the old collection so that we don't forget to update both places if we happen to add checkpoints before we refactor the rest of the code.

Reply:
Done.